### PR TITLE
zrok: support most of the arches

### PR DESCRIPTION
**Pull request by bbigras**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@bennyandresen

### CODE REVIEW

**Review by SuperSandro2000 — COMMENTED** [review]
Can't we compile from source instead?

**Comment by bbigras** [comment]
We tried but IIRC, it was a pita.

File: pkgs/tools/networking/zrok/default.nix [hunk 1]
```diff
@@ -1,15 +1,37 @@
...
     stripRoot = false;
-    sha256 = "sha256-gcmgpvfk7bciTmotTHObvZvLPdLudAR2vQneLKN+uE4=";
+    inherit sha256;
   };
 
+  updateScript = ./update.sh;
+
   installPhase = let
     interpreter = "$(< \"$NIX_CC/nix-support/dynamic-linker\")";
   in ''
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/zrok/default.nix:37]:
```
building '/nix/store/sgi21wc87b1n455zfi9dgm41pmlkfaiq-zrok-0.3.6.drv'...
unpacking sources
unpacking source archive /nix/store/gyfwhj5ma0dz61fsbmkmzv89s20y2jrx-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
no Makefile or custom buildPhase, doing nothing
installing
/nix/store/r46ydv8hlmx68nnclchjbgq0gjv6nr9n-stdenv-darwin/setup: line 1610: patchelf: command not found
error: builder for '/nix/store/sgi21wc87b1n455zfi9dgm41pmlkfaiq-zrok-0.3.6.drv' failed with exit code 127;
```

Comment (bbigras) [reply pkgs/tools/networking/zrok/default.nix:33]: Thanks. I dropped the darwin stuff since I don't have the hardware to test it.

**Comment by bbigras** [comment]
Thanks!

**Pull request merged**